src/goTest.ts

```diff
@@ -94,19 +94,19 @@ export function testCurrentPackage(timeout: string) {
  *
  * @param timeout a ParseDuration formatted timeout string for the tests.
  */
-export function testCurrentFile(timeout: string) {
+export function testCurrentFile(timeout: string, goConfig?: vscode.WorkspaceConfiguration): Thenable<boolean> {
 	let editor = vscode.window.activeTextEditor;
 	if (!editor) {
 		vscode.window.showInformationMessage('No editor is active.');
 		return;
 	}
-	getTestFunctions(editor.document).then(testFunctions => {
+	return getTestFunctions(editor.document).then(testFunctions => {
 		return goTest({
 			timeout: timeout,
 			dir: path.dirname(editor.document.fileName),
 			functions: testFunctions.map(func => { return func.name; })
-		});
-	}).then(null, err => {
```

Comment:
Put the `console.error(err)` to back where it was and returned promise from there